lib/messages/search-request.js

```diff
@@ -35,7 +35,7 @@ const isValidAttributeString = str => {
     return true
   }
   // ascii attribute names
-  if (/^[a-zA-Z][\w\d.;-]*$/.test(str) === true) {
+  if (/^[a-zA-Z][\w\d.;=*-]*$/.test(str) === true) {
```

Comment:
Why has the `*` character been added to the allowed characters?

Reply:
From my understanding the range attribute may end with an asterisk (*) to return all entries within a group up to a limit if that was set; so say you have 1525 members in a group and AD is set to limit only 1500 entries per request, you will get the first chunk of 1500 members with the following option `...;range=0-*`. Once you get that you can make another request like `...;range=1500-*` where AD would return the remaining 25 entries. I will mention though since we already know the limit that AD has set in our environment, we just go ahead and make our requests in range chunks of 1500 and when AD returns back the result with an asterisk as part of the range option, we know we got the last chunk of entries.

Reply:
So for an example if I was using ldapsearch, (please forgive all the removed text):
```
> ldapsearch ... '(&(objectClass=group)(cn=$staff-current))' 'member;range=0-*'
...
# filter: (&(objectClass=group)(cn=$staff-current))
# requesting: member;range=0-* 
...
member;range=0-1499: CN=staff1,OU=...
member;range=0-1499: CN=staff2,OU=...
...
member;range=0-1499: CN=staff1499,OU=...
member;range=0-1499: CN=staff1500,OU=...
...
```

Since there are only a total of 2396 current staff members we will get 896 more members in the next request, however, AD returns `member;range=1500-*` in place of `member;range=1500-2999` indicating that it has reached the end of the group.

```
> ldapsearch ... '(&(objectClass=group)(cn=$staff-current))' 'member;range=1500-*'
...
# filter: (&(objectClass=group)(cn=$staff-current))
# requesting: member;range=1500-* 
...
member;range=1500-*: CN=staff1501,OU=...
member;range=1500-*: CN=staff1502,OU=...
...
member;range=1500-*: CN=staff2395,OU=..
member;range=1500-*: CN=staff2396,OU=..
...
```

Reply:
So it is in the §3.1.1.3.1.3.3 spec after the following entry at the bottom of the page:
`This option takes the form
range=low-high`
on the next page of 162, use of the asterisk is mentioned:
`... Zero is used for low to specify the first entry. A client can substitute an asterisk for high to indicate all remaining entries (subject to any limitations imposed by the server on the maximum number of values to return).`

Reply:
Thank you for finding that. I was going to re-read the section before responding to see if I had missed anything. I clearly did.